test/test_processing_utils.py

```diff
@@ -404,3 +405,23 @@ async def test_json_data_not_moved_to_cache():
         )
         == data
     )
+
+
+@pytest.mark.parametrize(
+    "url",
+    [
+        "https://localhost",
+        "http://127.0.0.1/file/a/b/c",
+        "http://[::1]",
+        "https://192.168.0.1",
+        "http://10.0.0.1?q=a",
+        "https://172.16.0.1",
+    ],
+)
+def test_various_local_urls(url):
+    with pytest.raises(httpx.RequestError, match="Non-public IP address found"):
+        processing_utils.check_public_url(url)
+
+
+def test_valid_public_url():
+    assert processing_utils.check_public_url("https://www.example.com")
```

Comment:
Can we also test a valid public IP address works?

Reply:
imo would be cleaner to combine the two unit tests by adding a second argument to the test which is a boolean argument indicating whether the url should be allowed or not.

Reply:
what do you mean? this is an example of a domain that points to a public ip address that works. If you're saying a domain that points to a private address (so that this fails) that's very hard to set up, we'd need to set up a domain for that.